modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/StoreContext.js

```diff
@@ -30,6 +34,49 @@ function reducer(state = INITIAL_STATE, action) {
 		case ADD_WARNING:
 			nextState = state.warning ? state : {...state, warning: true};
 			break;
+		case SET_METRICS:
+			nextState = {
+				...state,
+				historicalReads: {
+					analyticsReportsHistoricalReads:
+						action.payload?.analyticsReportsHistoricalReads,
+				},
+				historicalViews: {
+					analyticsReportsHistoricalViews:
+						action.payload?.analyticsReportsHistoricalViews,
+				},
+				loading: false,
+				totalReads: action.payload?.analyticsReportsTotalReads,
+				totalViews: action.payload.analyticsReportsTotalViews,
+				trafficSources: action.payload.trafficSources,
```

Comment:
Why `payload?` on prior lines but `payload` here? Seems it should go everywhere, or nowhere.

Reply:
You mean access directly to the data without `payload` in the middle? `action.trafficSources`? Instead of using `{payload: metrics}` use `{...metrics}`?